File: AEDS/EXERCÍCIOS/NIVELAMENTO-RECURS/exercicio1.java
import java.util.Scanner;

public class exercicio1 {
    public static int contVogal(String str) {
        if (str.length() == 0) return 0;
        char c = str.charAt(0);
        if (c >= 'a' && c <= 'z') c = (char) (c - 32);
        if (c == 'A' || c == 'E' || c == 'I' || c == 'O' || c == 'U') return 1 + contVogal(str.substring(1));
        else return contVogal(str.substring(1));
    }
    public static int contConsoante(String str) {
        if (str.length() == 0) return 0;
        char c = str.charAt(0);
        if (c >= 'a' && c <= 'z') c = (char) (c - 32);
        if ((c >= 'A' && c <= 'Z') && !(c == 'A' || c == 'E' || c == 'I' || c == 'O' || c == 'U')) return 1 + contConsoante(str.substring(1));
        else return contConsoante(str.substring(1));
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str = sc.nextLine();
        System.out.println("Vogais: " + contVogal(str));
        System.out.println("Consoantes: " + contConsoante(str));
        sc.close();
    }
}

File: AEDS/EXERCÍCIOS/NIVELAMENTO-RECURS/exercicio1.c
#include <stdlib.h>
#include <stdio.h>
#include <ctype.h>

int contVogal (char *str) {
    str[0] = toupper(str[0]);
    if (str[0] == '\0') return 0;
    else if (str[0] == 'A' || str[0] == 'E' || str[0] == 'I' || str[0] == 'O' || str[0] == 'U') return 1 + contVogal(str + 1);
    else return contVogal(str + 1);
}

int contConsoante (char *str) {
    str[0] = toupper(str[0]);
    if (str[0] == '\0') return 0;
    else if ((str[0] >= 'A' && str[0] <= 'Z') && !(str[0] == 'A' || str[0] == 'E' || str[0] == 'I' || str[0] == 'O' || str[0] == 'U')) return 1 + contConsoante(str + 1);
    else return contConsoante(str + 1);
}

int main () {
    char str[100];
    scanf(" %[^\n]", str);
    printf("Vogais: %d\n", contVogal(str));
    printf("Consoantes: %d\n", contConsoante(str));
}
